BeeTracker/app/src/main/java/com/blueprint/foe/beetracker/Model/Submission.java

```diff
@@ -0,0 +1,172 @@
+        if (f > MAX_FACE) {
+            throw new MissingFormatArgumentException("Invalid face value");
+        }
+        face = f;
+    }
+
+    public int getFace() {
+        return face;
+    }
+
+    public void setThorax(int t) throws MissingFormatArgumentException {
+        if (t > MAX_THORAX) {
+            throw new MissingFormatArgumentException("Invalid thorax value");
+        }
+        thorax = t;
+    }
+
+    public int getThorax() {
+        return thorax;
+    }
+
+    public void setAbdomen(int a) throws MissingFormatArgumentException {
+        if (a > MAX_ABDOMEN) {
+            throw new MissingFormatArgumentException("Invalid abdomen value");
+        }
+        abdomen = a;
+    }
+
+    public int getAbdomen() {
+        return abdomen;
+    }
+
+    public void setHabitat(Habitat h) {
+        habitat = h;
+    }
+
+    public Habitat getHabitat() {
+        return habitat;
+    }
+
+    public void setWeather(Weather w) {
+        weather = w;
+    }
+
+    public Weather getWeather() {
+        return weather;
+    }
+
+    public void setLocation(Location location) {
+        this.location = location;
+    }
+
+    public Location getLocation() {
+        return location;
+    }
+
+    public Species getSpecies() throws MissingFormatArgumentException {
+        if (species == null) {
+            findSpecies();
+        }
+        return species;
+    }
+
+    public void findSpecies() throws MissingFormatArgumentException {
+        if (face < 0) {
+            throw new MissingFormatArgumentException("Face is missing");
+        } else if (thorax < 0) {
+            throw new MissingFormatArgumentException("Thorax is missing");
+        } else if (abdomen < 0) {
+            throw new MissingFormatArgumentException("Abdomen is missing");
+        } else if (face > MAX_FACE) {
+            throw new MissingFormatArgumentException("Invalid face value");
+        } else if (thorax > MAX_THORAX) {
+            throw new MissingFormatArgumentException("Invalid thorax value");
+        } else if (abdomen > MAX_ABDOMEN) {
+            throw new MissingFormatArgumentException("Invalid abdomen value");
+        }
+
+        if (face == 0 && thorax == 0 && abdomen == 0)
+            species = Species.Apidae;
+        else if (face == 0 && thorax == 0 && abdomen == 1)
+            species = Species.Andrenidae;
+        else if (face == 0 && thorax == 1 && abdomen == 0)
+            species = Species.Halictidae;
+        else if (face == 0 && thorax == 1 && abdomen == 1)
+            species = Species.Megachilidae;
+        else if (face == 1 && thorax == 0 && abdomen == 0)
+            species = Species.Colletidae;
+        else
+            species = Species.Unknown;
+    }
+
+    public boolean isComplete() {
+        return species != null && habitat != null && weather != null && location != null;
+    }
+
+    /* For debug only */
+    public void print() {
```

Comment:
Consider using `Log.d(...)` it's more android and you can log all kinds of things.